Source/CarthageKitTests/Fakes/FakeURLSession.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+final class FakeDownloadTask: NSURLSessionDownloadTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeDataTask: NSURLSessionDataTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeURLSession: NSURLSession {
```

Comment:
Additionally, I need to get you onboard with mocking, because on my list of refactors to make is injecting NSFileManager (probably with a `FileManager` protocol that `NSFileManager` conforms to) because Tests should not depend on the file structure at all. Furthermore, having the code littered with calls to `NSFileManager.defaultManager()` is bad practice. NSURLSession is only used in a single file, which made it much easier to get in.

I also have "remove calls to Task" on that list, because making calls to other processes is slow and untestable. However, that's much deeper in the stack of refactors I want to do.